### PR TITLE
Missile man

### LINKED ISSUE #17: Missile Man

### PR DESCRIPTION
**Pull request by Armaantakyar1**
everything viewable in the enemiesAndPlayerTest

closes #17 Missile man 
also made enemy and projectile base classes 

the missile is homing till it reaches a certain range and then it goes in a straight line in a faster speed. I did this because it was impossible for the player to dodge the bullet.

I plan to make a pickable object but I need Hassan's updated telekinesis stuff

### CODE REVIEW

**Review by avinash751 — APPROVED** [review]
code is straight forward and the  functionality works pretty well. the only feedback i have is with the jitter when the bullet starts targeting the player. code wise it is better to put missile stats in missile man and make the stats transfer from missile man to the missle when the missile is instantiated. this way it's easier to balance prefabs. its a similar thing that was done for bullet for the turret, i will still approve this merge since its very functional and changes to be made our small. better to implement these changes next sprint. coo stuff

**Review by devTrivedi1 — APPROVED** [review]
Looks good to me so far, will see how it goes when I put it in the actual levels out!

File: Assets/PROJECT/Scripts/Enemies/missileMan.cs
Comment (HassanAbusalih) [inline Assets/PROJECT/Scripts/Enemies/missileMan.cs:1]: This script isn't named properly

**Review by HassanAbusalih — CHANGES_REQUESTED** [review]
I think it'd be best if you merged dev into your branch and set up resetting first, since there might end up being a bug related to it. I partially set it up for testing, but you've obviously worked on it more since.

**Comment by avinash751** [comment]
> I think it'd be best if you merged dev into your branch and set up resetting first, since there might end up being a bug related to it. I partially set it up for testing, but you've obviously worked on it more since.

yeah agreed better to do that

**Pull request merged**